plugin/pulpcore/plugin/download/asyncio/file.py

```diff
@@ -0,0 +1,46 @@
+import os
+from urllib.parse import urlparse
+
+from .base import BaseDownloader, DownloadResult
+
+
+class FileDownloader(BaseDownloader):
+    """
+    A downloader for reading files from the filesystem.
+
+    It provides digest and size validation along with computation of the digests needed to save the
+    file as an Artifact.
+    """
+
+    def __init__(self, uri, **kwargs):
+        """
+        Download files from a uri that starts with `file://`
+
+        Args:
+            uri (str): The uri to the file. This is expected to begin with `file://`
+            kwargs (dict): This accepts the parameters of
+                :class:`~pulpcore.plugin.download.asyncio.BaseDownloader`.
+        """
+        self.uri = uri
+        kwargs['custom_file_object'] = open(os.devnull, 'w')
```

Comment:
Why are you writing the downloaded file to /dev/null?

Reply:
It's because I didn't want to have the file importer make a copy of the file. I realize now that it needs to make a copy because otherwise when the original file is moved into place it will destroy the original file:// data. I'm going to remove this code now so that it makes a copy instead that way the original data can be imported multiple times.